catalyst/dl/callbacks/cmc_callback.py

```diff
@@ -0,0 +1,209 @@
+from typing import List
+
+import torch
+
+from catalyst.core import IRunner
+from catalyst.core.callback import CallbackOrder
+from catalyst.dl import Callback
+from catalyst.dl.callbacks.metrics.accuracy import get_default_accuracy_args
+from catalyst.utils.metrics.cmc_score import cmc_score
+
+
+class CMCScoreCallback(Callback):
+    """
+    Cumulative Matching Characteristics callback
+
+    You should use it with `ControlFlowCallback`
+    and add all query/gallery sets to loaders.
+    Loaders should contain "is_query" and "label" key.
+
+    .. code-block:: python
+        import os
+        import torch
+        from torch.utils.data import DataLoader
+
+        from catalyst.contrib.nn.criterion import TripletLoss
+        from catalyst.core.callbacks import ControlFlowCallback
+        from catalyst.dl import CMCScoreCallback, SupervisedRunner
+        from catalyst.contrib.datasets import MNIST, MnistQGDataset
+        from catalyst.contrib.data.transforms import ToTensor
+
+        train = MNIST(
+            os.getcwd(),
+            download=True,
+            train=True,
+            transform=ToTensor()
+        )
+        valid = MNIST(
+            os.getcwd(),
+            download=True,
+            train=False,
+            transform=ToTensor()
+        )
+        query_gallery = MnistQGDataset(os.getcwd(), transform=ToTensor())
+
+        train_loader = DataLoader(train, batch_size=32, shuffle=True)
+        valid_loader = DataLoader(valid, batch_size=32)
+        query_gallery = DataLoader(query_gallery, batch_size=64)
+
+        loaders = {
+            "train": train_loader,
+            "valid": valid_loader,
+            "valid_qg": query_gallery
+        }
+
+        callbacks = [
+            ControlFlowCallback(
+                base_callback=CMCScoreCallback(
+                    topk_args=[1, 3, 5]
+                ),
+                loaders="valid_qg"
+            )
+        ]
+
+        criterion = TripletLoss()
+
+        class Flatten(torch.nn.Module):
```

Comment:
we already have it :)
https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/nn/modules/common.py#L8